src/Dev/Deprecation.php

```diff
@@ -67,12 +67,22 @@ class Deprecation
      * Must be configured outside of the config API, as deprecation API
      * must be available before this to avoid infinite loops.
      *
-     * This will be overriden by the SS_DEPRECATION_ENABLED environment if present
+     * This will be overriden by the SS_DEPRECATION_ENABLED environment variable if present
      *
      * @internal - Marked as internal so this and other private static's are not treated as config
      */
     private static bool $currentlyEnabled = false;
 
+    /**
+     * @internal
+     */
+    private static bool $shouldSuppressForHttp = true;
```

Comment:
```suggestion
    private static bool $showForHttp = false;
```

This seems a bit back-to-front being "suppress", seems the logic should be the other way around and be "show", which is simpler to understand and to how we have "showNoReplacementNotices" below in this file.

Also no need to have the word "should"

Reply:
Should also add something to the phpdoc saying that that this is only for outputting to the browser, it'll always log no matter the value here

Reply:
It's not "for the browser" (though that's how it's framed in the docs for simplicity" - it's for _any_ http request. But I'll make an explicit point about it not affecting logs.

Reply:
I had it all as suppress because the logic does actually end up being simpler IMO:

```php
// suppress if everything is true - nothing needs to be negated
protected function shouldSuppressError(int $errorCode): bool
{
    return $errorCode === E_USER_DEPRECATED
        && Deprecation::triggeringError()
        && (Director::is_cli() ? Deprecation::shouldShowForCli() : Deprecation::shouldShowForHttp());
}

// show if any given thing is false - requires negating almost everything.
protected function shouldShowError(int $errorCode): bool
{
    return $errorCode !== E_USER_DEPRECATED
        || !Deprecation::triggeringError()
        || (Director::is_cli() ? Deprecation::shouldShowForCli() : Deprecation::shouldShowForHttp());
}

// And then in `write()` we go from
if ($this->shouldSuppressError($errorCode)) {
    return;
}
// to
if (!$this->shouldShowError($errorCode)) { // negating again
    return;
}
```

That said, if it makes a more intuitive API then that's more important, so I've made the change on the assumption you're correct about what will be more intuitive for developers generally.

Reply:
Also added a unit test which should make sure the logic works as expected